relay/pkg/block/push.go

```diff
@@ -0,0 +1,214 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+)
+
+// pullHeadersFromQueue waits until we have `headersBatchSize` headers from
+// the queue or until the queue fails to yield a header for
+// `headerTimeout` duration.
```

Comment:
`... and returns them from the function`?

Reply:
Please see https://github.com/keep-network/tbtc/commit/c6fb5d71e9c976fb98ee676e36dfcd48ff6221fe